src/WebJobs.Script/Config/ScriptSettingsManager.cs

```diff
@@ -153,6 +154,24 @@ public virtual void SetSetting(string settingKey, string settingValue)
             }
         }
 
+        private IConfiguration InitializeConfiguration(Lazy<IConfiguration> currentConfiguration)
+        {
+            var configuration = _configurationFactory();
+
+            // If the factory returned a cached instance
+            // that is the same as the instance we previously had,
+            // reload the configuration on that instance.
```

Comment:
Recommend adding some comments indicating why we're doing this - i.e. to enable us to pick up environment settings that change at runtime.

Reply:
Thought about that, this really applies to all configuration, though. Not just app settings.